Remove deprecated collation types "big5han" and "gb2312" from JSTests expectations

shvaikalesh opened the pull request:
#### bc9716f4ee6360017237a4b85426cc3b56fd8b66
<pre>
Remove deprecated collation types &quot;big5han&quot; and &quot;gb2312&quot; from JSTests expectations
<a href="https://bugs.webkit.org/show_bug.cgi?id=268879">https://bugs.webkit.org/show_bug.cgi?id=268879</a>
&lt;<a href="https://rdar.apple.com/problem/122437088">rdar://problem/122437088</a>&gt;

Reviewed by Yusuke Suzuki and Mark Lam.

These collation types were deprecated for a long time and not available in Chrome / Firefox [1],
and finally removed in ICU 74 [2].

[1]: <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getCollations#supported_collation_types">https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getCollations#supported_collation_types</a>
[2]: <a href="https://github.com/unicode-org/icu/pull/2365">https://github.com/unicode-org/icu/pull/2365</a>

* JSTests/stress/intl-enumeration.js:
* JSTests/stress/intl-locale-info.js:

Canonical link: <a href="https://commits.webkit.org/274236@main">https://commits.webkit.org/274236@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/19fe9610b88eaa74f1fdfac54ca9b7cec2178625

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38226 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17167 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40541 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40783 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/34006 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19918 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/14508 "Built successfully") | [⏳ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/WPE-WK2-Tests-EWS "Waiting to run tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38799 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14485 "Passed tests") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12612 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12584 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42060 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/31845 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34741 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34618 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38433 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/37905 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13176 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10865 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36624 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14728 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/44848 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13592 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/9178 "Found 3 new JSC stress test failures: wasm.yaml/wasm/gc/bug266056.js.wasm-eager, wasm.yaml/wasm/gc/bug266127.js.wasm-eager, wasm.yaml/wasm/gc/call_ref.js.wasm-eager (failure)") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4996 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/14192 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (changes requested):
It needs to be changed to work with both cases since bots using old OS will not have the latest ICU.

MenloDorian reviewed (approved):
r=me

Constellation reviewed (approved):
r=me

Commit:
Remove deprecated collation types "big5han" and "gb2312" from JSTests expectations
https://bugs.webkit.org/show_bug.cgi?id=268879
<rdar://problem/122437088>

Reviewed by Yusuke Suzuki and Mark Lam.

These collation types were deprecated for a long time and not available in Chrome / Firefox [1],
and finally removed in ICU 74 [2].

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getCollations#supported_collation_types
[2]: https://github.com/unicode-org/icu/pull/2365

* JSTests/stress/intl-enumeration.js:
* JSTests/stress/intl-locale-info.js:

Canonical link: https://commits.webkit.org/274236@main
